File: src/main/java/me/glorantq/aboe/chat/common/PacketUtils.java
package me.glorantq.aboe.chat.common;

import io.netty.buffer.ByteBuf;

import java.nio.charset.Charset;

public class PacketUtils {
    public static ByteBuf writeString(ByteBuf buffer, String value) {
        buffer.writeInt(value.length());
        buffer.writeBytes(value.getBytes());

        return buffer;
    }

    public static String readString(ByteBuf buffer) {
        int length = buffer.readInt();
        byte[] data = new byte[length];

        buffer.readBytes(data);

        return new String(data, Charset.forName("UTF-8"));
    }
}

File: src/main/kotlin/me/glorantq/aboe/chat/common/PacketUtils.kt
package me.glorantq.aboe.chat.common

import io.netty.buffer.ByteBuf

import java.nio.charset.Charset

object PacketUtils {
    @JvmStatic fun writeString(buffer: ByteBuf, value: String): ByteBuf {
        buffer.writeInt(value.length)
        buffer.writeBytes(value.toByteArray())

        return buffer
    }

    @JvmStatic fun readString(buffer: ByteBuf): String {
        val length = buffer.readInt()
        val data = ByteArray(length)

        buffer.readBytes(data)

        return String(data, Charset.forName("UTF-8"))
    }
}
